autosklearn/evaluation/train_evaluator.py

```diff
@@ -654,6 +703,33 @@ def fit_predict_and_loss(self, iterative: bool = False) -> None:
                     if Y_optimization_pred[i] is not None
                 ]
             )
+            if isinstance(X_targets[0], np.ndarray):
```

Comment:
There's now quite a bit of code duplicated, maybe it should go into its own function?

Reply:
agreed and done